text/0000-more-promise-methods.md

```diff
@@ -0,0 +1,63 @@
+- Feature Name: more-promise-methods
+- Start Date: 2017-02-25
+- RFC PR: 
+- Pony Issue: 
+
+# Summary
+
+This RFC proposes adding additional methods to the Promise actor.
+
+# Motivation
+
+The Promise actor is a useful tool for representing a value before it is available, though the public API is a bit sparse. Only 3 basic behaviors and functions are currently available to the user (apply, reject, and next). The following additional methods will improve composability and reflect common operations found in other promises/futures libraries.
+
+# Detailed design
+
+```pony
+
+fun tag join[B: Any #share = A](p: Promise[B]): Promise[(A, B)]
+	"""
+	Join two promises into one promise that returns the result of both when they
+	are fulfilled. If either of the promises is rejected then the new promise is
+	also rejected.
+	"""
+
+fun tag many(ps: Array[Promise[A]] val): Promise[Array[A] val]
```

Comment:
I think this should be `new` rather than `fun tag` - using `fun tag` implies that the receiver promise will be part of the result, but I think that makes for an awkward API.  In my opinion, the passed array of promises should be the entire array expected to get the results of.